python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9316,3 +9316,67 @@ def test_dataframe_eval(df_eval, expr, dtype):
 def test_dataframe_eval_errors(df_eval, expr):
     with pytest.raises(ValueError):
         df_eval.eval(expr)
+
+
+@pytest.mark.parametrize(
+    "gdf,subset",
+    [
+        (
+            cudf.DataFrame(
+                {"num_legs": [2, 4, 4, 6], "num_wings": [2, 0, 0, 0]},
+                index=["falcon", "dog", "cat", "ant"],
+            ),
+            ["num_legs"],
+        ),
+        (
+            cudf.DataFrame(
+                {
+                    "first_name": ["John", "Anne", "John", "Beth"],
+                    "middle_name": ["Smith", None, None, "Louise"],
+                }
+            ),
+            ["first_name"],
+        ),
+    ],
+)
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("normalize", [True, False])
+@pytest.mark.parametrize("dropna", [True, False])
+@pytest.mark.parametrize("use_subset", [True, False])
+def test_value_counts(
+    gdf,
+    subset,
+    sort,
+    ascending,
+    normalize,
+    dropna,
+    use_subset,
+):
+    pdf = gdf.to_pandas()
+
+    got = gdf.value_counts(
+        subset=subset if (use_subset) else None,
+        sort=sort,
+        ascending=ascending,
+        normalize=normalize,
+        dropna=dropna,
+    )
+    expected = pdf.value_counts(
+        subset=subset if (use_subset) else None,
+        sort=sort,
+        ascending=ascending,
+        normalize=normalize,
+        dropna=dropna,
+    )
+
+    if not dropna:
+        # Convert the Pandas series to a cuDF one due to difference
+        # in the handling of NaNs between the two (<NA> in cuDF and
+        # NaN in Pandas) when dropna=False.
+        assert_eq(got, cudf.from_pandas(expected))
+    else:
+        assert_eq(got, expected)
```

Comment:
```suggestion
        assert_eq(got.sort_index(), cudf.from_pandas(expected).sort_index())
    else:
        assert_eq(got.sort_index(), expected.sort_index())
```

We can switch this to use `sort_index` since we don't have to always be consistent to pandas in cases of a groupby like @isVoid pointed out.  But for this to work we will need to merge: https://github.com/rapidsai/cudf/pull/10957

Reply:
Now that #10957 is merged you should be able to apply @galipremsagar's suggestion.